### PR TITLE
Feature/76 mapping lab value

### PR DESCRIPTION
**Pull request by jsachau**
Please check the TODOS in the `FhirDiagnosticReportOpenehrLabResults` class as i had some questions concerning this mapping.

Apart from that the mapping is updated.

### CODE REVIEW

**Comment by jsachau** [comment]
@ppazos The robot tests are failing while the FHIR bridge tests pass as expected. Can you identify the error?

**Comment by jsachau** [comment]
@ppazos @wlad, it replace the previous one.

**Comment by wlad** [comment]
@jsachau @ppazos  I pushed a fix
- CI now stores robot output under artifacts tab to allow better investigation of test failure
  - https://app.circleci.com/pipelines/github/ehrbase/fhir-bridge/351/workflows/877d1860-6bc0-4f60-9fa6-5811b33a7a57/jobs/480/artifacts
  - https://480-257906436-gh.circle-artifacts.com/0/00_robot_results/log.html
- updated profile url in failing test

image

NOTE: CI is now failing because of Sonar complainig about some minor issue. Let me know if it can be ignored then I'll resolve it as `won't fix` 
image

**Comment by jsachau** [comment]
@wlad: The  _Laborbefund.opt_ has been replaced by _GECCO_Laborbefund.opt_  and the Profile has also changed. It has a new URL: [https://www.medizininformatik-initiative.de/fhir/core/modul-labor/StructureDefinition/ObservationLab](https://www.medizininformatik-initiative.de/fhir/core/modul-labor/StructureDefinition/ObservationLab)

**Comment by jsachau** [comment]
Concerning the Sona complaints: I updated the code in [](https://github.com/ehrbase/fhir-bridge/blob/develop/src/main/java/org/ehrbase/fhirbridge/rest/OpenEhrController.java) so that it uses the new GECCO_Laborbefund composition instead of the old one. In the old code (see link), the parameter body was reused. I assume that the new composition has to be placed in the body variable and thus I would mark the complaint as "Resolve as won't fix"

**Comment by ppazos** [comment]
@wlad I think the error you see is because of the GET (Read operation) wasn't updated, I guess from the last comment of @jsachau that is updated now.

The only thing I see failing is the SonarCloud quality check https://sonarcloud.io/dashboard?id=ehrbase_fhir-bridge&pullRequest=85

How can we improve that @wlad ?

**Comment by ppazos** [comment]
@jsachau I tried to solve the conflicts and Sonar is complaining about this code that has some tests that were not removed https://sonarcloud.io/project/issues?id=ehrbase_fhir-bridge&open=AXV7rTXuavZxGRZp53xZ&pullRequest=85&resolved=false&types=BUG

I think fixing that will also raise the coverage that is know under the required limit https://github.com/ehrbase/fhir-bridge/pull/85/checks?check_run_id=1355136088

**Comment by wlad** [comment]
@jsachau @ppazos I marked the sonar issue as  "Resolve as won't fix". PR is clear and ready to be merged now.

File: src/main/java/org/ehrbase/fhirbridge/rest/OpenEhrController.java [hunk 1]
```diff
@@ -44,43 +43,37 @@ public OpenEhrController(EhrbaseService service) {
     @PostMapping(path = "/{ehr_id}/labor")
     public ResponseEntity<VersionUid> postLabor(
             @PathVariable(value = "ehr_id") UUID ehrId,
-            @RequestBody LaborbefundComposition body) {
+            @RequestBody GECCOLaborbefundComposition body) {
 
         // TESTING      TODO: remove
```
Comment (ppazos) [inline src/main/java/org/ehrbase/fhirbridge/rest/OpenEhrController.java:48]: @jsachau should something here be removed?

**Comment by ppazos** [comment]
Thanks @wlad I just a the question here https://github.com/ehrbase/fhir-bridge/blob/461721250632e41b95a325582ad2cfb7bd19f40f/src/main/java/org/ehrbase/fhirbridge/rest/OpenEhrController.java#L49 for @jsachau because seems something needs to be removed.

**Comment by jsachau** [comment]
@ppazos the TODO was already in the code before. I didn't write and change it.

**Comment by jakesmolka** [comment]
This was my comment (you can check the 'git blame') from when I wrote the initial REST controller. The setting of all those parameters there was just to make sure everything worked technically with the given set of examples at that time (also in connection with EHRbase). If you two think those are correct default values you can remove the comment block.

But while something like

``` java
        body.setLanguage(Language.DE);
        body.setCategoryDefiningcode(CategoryDefiningcode.EVENT);
        body.setStartTimeValue(OffsetDateTime.now());
        body.setSettingDefiningcode(SettingDefiningcode.NURSING_HOME_CARE);
        body.setTerritory(Territory.DE);
```
probably makes sense as a default - the following is most likely very wrong (as it would even overwrite the given magnitude from the input)

``` java
        LaboranalytResultatAnalytResultatDvquantity result_value = new LaboranalytResultatAnalytResultatDvquantity();
        result_value.setAnalytResultatMagnitude(11.00);
        result_value.setAnalytResultatUnits("mg");
```

I don't know what this PR is about, but I'd guess changing the REST controller would be out of scope, so maybe let's just create a new issue for that? :slightly_smiling_face:

**Comment by wlad** [comment]
> ... but I'd guess changing the REST controller would be out of scope, so maybe let's just create a new issue for that?
@ppazos  @jsachau  please remeber to create a dedicated issue for that if you think it is needed.

I'm going to merge this PR now to avoid merge conflicts in other PRs that I want to merge next.

**Comment by jakesmolka** [comment]
@ppazos @jsachau I couldn't find an issue for what I described. I created one, because I think this is a bug. If you created one yourself already please link/mention it in mine.

**Comment by ppazos** [comment]
thanks @jakesmolka I'll take a look and create issues accordingly

**Pull request merged**